spark/spark-base/src/test/java/com/thoughtworks/go/api/base/JsonUtils.java

```diff
@@ -34,6 +34,14 @@ public static String toObjectString(Consumer<OutputWriter> consumer) {
         return new JsonOutputWriter(new StringWriter(1024), new TestRequestContext()).forTopLevelObject(consumer).writer.toString();
     }
 
+    public static String toObjectStringWithoutLinks(Consumer<OutputWriter> consumer) {
+        return new JsonOutputWriter(new StringWriter(1024), null).forTopLevelObject(consumer).writer.toString();
+    }
+
+    public static Map<String, ?> toObjectWithoutLinks(Consumer<OutputWriter> consumer) throws IOException {
+        return JsonOutputWriter.OBJECT_MAPPER.readValue(toObjectStringWithoutLinks(consumer), Map.class);
+    }
+
```

Comment:
I noticed that we're not sending links anymore in error responses for ArtifactStoreConfig and Roles. Is this intentional? Seems minor, but I wonder if it would affect any 3rd parties consuming the API. Even if it doesn't warrant a version bump, we should probably mention it in the release notes.

Reply:
@akshaydewan

We do include links still AFAIK. The place where we remove links is when errors happen and we return an error message with a data key, holding the payload. I don’t think there’s any documentation for that behavior? It doesn’t make sense to send links back within an error message IMO.

But do let me know if I’m mistaken and the links are *not* being returned on GET and INDEX. That would be unintentional.

Reply:
To be clear, I’m referring to these error messages:
```
{
  message: “error occurred!”,
  data: {
    // request payload goes here, no need to include links because it’s just dumping what the user sent
  }
}
```

Reply:
It's more a matter of consistency IMO. And in case of errors,  users may still find the `doc` link helpful.

Reply:
We can certainly document this change if the old behavior is documented at all. I’ll look to see if we do.